yt/frontends/chimera/data_structures.py

```diff
@@ -0,0 +1,334 @@
+                            connectivity[(t - 1) * (nzd - 1) : t * (nzd - 1)] + nzd
+                        )
+                for r in range(
+                    nxd - 1
+                ):  # Increments this shell along r to define a sphere
+                    if r > 0:
+                        connectivity[r * (nyzd_) : (r + 1) * (nyzd_)] = (
+                            connectivity[(r - 1) * (nyzd_) : r * (nyzd_)] + nyzd
+                        )
+
+                mesh = ChimeraMesh(
+                    n, self.index_filename, connectivity, coords, self
+                )  # Creates a mesh object
+
+                if "grid_" in file:
+                    print(f"Mesh {n+1}/{len(index_filenames)} generated")
+                    self.meshes.append(
+                        mesh
+                    )  # Adds new mesh to the list of generated meshes
+
+    def _detect_output_fields(self):  # Reads in the available data fields
+        with h5py.File(self.index_filename, "r") as f:
+            fluids = [
+                ("chimera", i)
+                for i in f["fluid"]
+                if np.shape(f["fluid"][i]) == np.shape(f["fluid"]["rho_c"])
+            ]
+            abundance = [
+                ("chimera", i)
+                for i in f["abundance"]
+                if np.shape(f["abundance"][i]) == np.shape(f["fluid"]["rho_c"])
+            ]
+            e_rms = [("chimera", f"e_rms_{i+1}") for i in range(4)]
+            lumin = [("chimera", f"lumin_{i+1}") for i in range(4)]
+            num_lumin = [("chimera", f"num_lumin_{i+1}") for i in range(4)]
+            a_name = [("chimera", i.decode("utf-8")) for i in f["abundance"]["a_name"]]
+            self.field_list = (
+                fluids
+                + abundance
+                + e_rms
+                + lumin
+                + num_lumin
+                + [("chimera", "abar"), ("chimera", "shock")]
+                + a_name
+            )
+            if np.shape(f["abundance"]["nse_c"]) != np.shape(f["fluid"]["rho_c"]):
+                self.field_list += [("chimera", "nse_c")]
+
+    def _chunk_io(self, dobj, cache=True, local_only=False):  # Creates Data chunk
+        gobjs = getattr(dobj._current_chunk, "objs", dobj._chunk_info)
+        for subset in gobjs:
+            yield YTDataChunk(
+                dobj, "io", [subset], self._count_selection(dobj, [subset]), cache=cache
+            )
+
+    def _setup_data_io(self):
+        self.io = io_registry[self.dataset_type](self.dataset)
+
+
+class ChimeraDataset(Dataset):
+    _index_class = ChimeraUNSIndex  # ChimeraHierarchy
+    _field_info_class = ChimeraFieldInfo
+
+    def __init__(
+        self,
+        filename,
+        dataset_type="chimera",
+        storage_filename=None,
+        units_override=None,
+    ):
+        # refinement factor between a grid and its subgrid
+        self.refine_by = 2  # Somewhat superfluous for Chimera, but left to avoid errors
```

Comment:
the current convention to signal that this value is inert is to set it to 1
```suggestion
        self.refine_by = 1  # Somewhat superfluous for Chimera, but left to avoid errors
```

Reply:
This was originally set to 2 early in development because using 1 prompted an error, but it seems this error no longer occurs. Change implemented in 10df6b4.